tests/client/amo/sagas/testCategories.js

```diff
@@ -0,0 +1,119 @@
+} from 'core/constants';
+import apiReducer from 'core/reducers/api';
+import categoriesReducer from 'core/reducers/categories';
+import { signedInApiState } from 'tests/client/amo/helpers';
+
+
+describe('categoriesSaga', () => {
+  let initialState;
+  let sagaTester;
+  let store;
+
+  beforeEach(() => {
+    store = createStore().store;
+    const state = store.getState();
+    initialState = {
+      api: signedInApiState,
+      categories: state.categories,
+    };
+
+    sagaTester = new SagaTester({
+      initialState,
+      reducers: { api: apiReducer, categories: categoriesReducer },
+    });
+    sagaTester.start(categoriesSaga);
+  });
+
+  it('should get Api from state then make API request to categories', async () => {
+    const mockApi = sinon.mock(api);
+    const entities = sinon.stub();
+    const result = sinon.stub();
+
+    mockApi
+      .expects('categories')
+      .once()
+      .withArgs({
+        api: { ...initialState.api },
+      })
+      .returns(Promise.resolve({ entities, result }));
+
+    assert.deepEqual(sagaTester.getState(), initialState);
+
+    sagaTester.dispatch(actions.categoriesFetch());
+
+    assert.deepEqual(sagaTester.getState(), {
+      ...initialState,
+      categories: { ...initialState.categories, loading: true },
+    });
+
+    await sagaTester.waitFor(CATEGORIES_LOAD);
+
+    const calledActions = sagaTester.getCalledActions();
+
+    // First action is CATEGORIES_FETCH.
+    assert.deepEqual(calledActions[0], actions.categoriesFetch());
+
+    // Next action is showing the loading bar.
+    assert.deepEqual(calledActions[1], showLoading());
+
+    // Next action is loading the categories returned by the API.
+    assert.deepEqual(calledActions[2], actions.categoriesLoad({ result }));
+
+    // Last action is to hide the loading bar.
+    assert.deepEqual(calledActions[3], hideLoading());
+
+    mockApi.verify();
+  });
+
+  it('should dispatch fail if API request fails', async () => {
+    const mockApi = sinon.mock(api);
+    const error = sinon.stub();
+
+    mockApi
+      .expects('categories')
+      .once()
+      .withArgs({
+        api: { ...initialState.api },
+      })
+      .throws(error);
+
+    assert.deepEqual(sagaTester.getState(), initialState);
+
+    sagaTester.dispatch(actions.categoriesFetch());
+
+    await sagaTester.waitFor(CATEGORIES_FAIL);
+
+    const calledActions = sagaTester.getCalledActions();
+
+    // First action is CATEGORIES_FETCH.
+    assert.deepEqual(calledActions[0], actions.categoriesFetch());
+
+    // Next action is showing the loading bar.
+    assert.deepEqual(calledActions[1], showLoading());
+
+    // Next action is loading the categories returned by the API.
+    assert.deepEqual(calledActions[2], actions.categoriesFail(error));
+
+    // Last action is to hide the loading bar.
+    assert.deepEqual(calledActions[3], hideLoading());
+
+    mockApi.verify();
+  });
+
+  it('should yield takeEvery() for the main generator', () => {
```

Comment:
It ensures we response to all category actions instead of just one. Though actually because of how the categories API works it might be better to just have one request... I'll see.